libraries/ms-drivers/test/test_max6643_fan_controller.c

```diff
@@ -0,0 +1,99 @@
+void test_max6643_init_works(void) {
+  GpioAddress test_fanfail_pin = { .port = GPIO_PORT_A, .pin = 0 };
+  GpioAddress test_overtemp_pin = { .port = GPIO_PORT_A, .pin = 1 };
+
+  Max6643Settings valid_settings = { .fanfail_pin = test_fanfail_pin,
+                                     .overtemp_pin = test_overtemp_pin,
+                                     .fanfail_callback = NULL,
+                                     .overtemp_callback = NULL,
+                                     .fanfail_callback_context = NULL,
+                                     .overtemp_callback_context = NULL };
+
+  s_mock_storage.fanfail_pin = valid_settings.fanfail_pin;
+  s_mock_storage.overtemp_pin = valid_settings.overtemp_pin;
+  s_mock_storage.fanfail_callback = valid_settings.fanfail_callback;
+  s_mock_storage.overtemp_callback = valid_settings.overtemp_callback;
+  s_mock_storage.fanfail_callback_context = valid_settings.fanfail_callback_context;
+  s_mock_storage.overtemp_callback_context = valid_settings.overtemp_callback_context;
+
+  TEST_ASSERT_OK(max6643_init(&s_storage, &valid_settings));
+
+  TEST_ASSERT_EQUAL(s_mock_storage.fanfail_pin.pin, s_storage.fanfail_pin.pin);
+  TEST_ASSERT_EQUAL(s_mock_storage.fanfail_pin.port, s_storage.fanfail_pin.port);
+  TEST_ASSERT_EQUAL(s_mock_storage.overtemp_pin.pin, s_storage.overtemp_pin.pin);
+  TEST_ASSERT_EQUAL(s_mock_storage.overtemp_pin.port, s_storage.overtemp_pin.port);
+  TEST_ASSERT_EQUAL(s_mock_storage.fanfail_callback, s_storage.fanfail_callback);
+  TEST_ASSERT_EQUAL(s_mock_storage.fanfail_callback_context, s_storage.fanfail_callback_context);
+  TEST_ASSERT_EQUAL(s_mock_storage.overtemp_callback, s_storage.overtemp_callback);
+  TEST_ASSERT_EQUAL(s_mock_storage.overtemp_callback_context, s_storage.overtemp_callback_context);
+}
+
+// test that interrupts work
+void test_max6643_interrupts(void) {
+  GpioAddress test_fanfail_pin = { .port = GPIO_PORT_A, .pin = 0 };
+  GpioAddress test_overtemp_pin = { .port = GPIO_PORT_A, .pin = 1 };
+
+  Max6643Settings valid_settings = {
+    .fanfail_pin = test_fanfail_pin,
+    .overtemp_pin = test_overtemp_pin,
+    .fanfail_callback = (GpioItCallback)prv_test_max6643_fanfail_interrupt_callback,
+    .overtemp_callback = (GpioItCallback)prv_test_max6643_overtemp_interrupt_callback,
+    .fanfail_callback_context = NULL,
+    .overtemp_callback_context = NULL
+  };
+
+  s_mock_storage.fanfail_pin = valid_settings.fanfail_pin;
+  s_mock_storage.overtemp_pin = valid_settings.overtemp_pin;
+  s_mock_storage.fanfail_callback = valid_settings.fanfail_callback;
+  s_mock_storage.overtemp_callback = valid_settings.overtemp_callback;
+
+  // add callback to storage
+  TEST_ASSERT_OK(max6643_init(&s_storage, &valid_settings));
+
+  // trigger interrupt and fetch data
+  s_mock_statuses.FANFAIL_INTERRUPT_STATUS = 1;
+  gpio_it_trigger_interrupt(&test_fanfail_pin);
+
+  s_mock_statuses.OVERTEMP_INTERRUPT_STATUS = 1;
+  gpio_it_trigger_interrupt(&test_overtemp_pin);
```

Comment:
This test will pass even if the callbacks aren't called at all! I'd recommend making the callbacks increment some static variable like `s_times_fanfail_callback_called` and checking that after the `gpio_it_trigger_interrupt` call.